auto_round/utils.py

```diff
@@ -1265,6 +1265,144 @@ def _gguf_args_check(args):
 
     return args
 
+
+def _to_model_dtype(model, model_dtype):
+    if model_dtype != None:
+        try:
+            if model_dtype == "float16" or model_dtype == "fp16":
+                model = model.to(torch.float16)
+            elif model_dtype == "bfloat16" or model_dtype == "bfp16" or model_dtype == "bf16":
+                model = model.to(torch.bfloat16)
+            elif model_dtype == "float32" or model_dtype == "fp32":
+                model = model.to(torch.float32)
+        except:
+            logger.error("please use more device to fit the device or just use one device")
```

Comment:
Help split this into two functions later. The first should be a meta-device checker: if any parameters are on the meta device, log an error if it's unsupported, or skip model dtype conversion if it is supported.